### ISSUE TITLE
PyQt6 6.1 flag -> enum possible breaking change

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#1692]
PyQt6 6.1 will rename flag name (e.g. QtCore.Qt.Orientation**s**) to enum name (e.g. QtCore.Qt.Orientation).

See https://www.riverbankcomputing.com/pipermail/pyqt/2021-March/043713.html
and development snapshot Changelog at https://riverbankcomputing.com/software/pyqt/download

This would be a breaking change for the generated ui code, e.g.:
```bash
$ git grep Orientations
examples/exampleLoaderTemplate_pyqt6.py:        self.splitter.setOrientation(QtCore.Qt.Orientations.Horizontal)
pyqtgraph/canvas/CanvasTemplate_pyqt6.py:        self.splitter.setOrientation(QtCore.Qt.Orientations.Horizontal)
pyqtgraph/canvas/CanvasTemplate_pyqt6.py:        self.vsplitter.setOrientation(QtCore.Qt.Orientations.Vertical)
pyqtgraph/console/template_pyqt6.py:        self.splitter.setOrientation(QtCore.Qt.Orientations.Vertical)
pyqtgraph/graphicsItems/PlotItem/plotConfigTemplate_pyqt6.py:        self.gridAlphaSlider.setOrientation(QtCore.Qt.Orientations.Horizontal)
pyqtgraph/graphicsItems/PlotItem/plotConfigTemplate_pyqt6.py:        self.alphaSlider.setOrientation(QtCore.Qt.Orientations.Horizontal)
pyqtgraph/imageview/ImageViewTemplate_pyqt6.py:        self.splitter.setOrientation(QtCore.Qt.Orientations.Vertical)
```

What should be the way forward for pyqtgraph if the old name (Orientations) were removed in PyQt6 6.1?
1) drop support for PyQt6 < 6.1 from a particular commit?

**Comment by j9ac9k** [comment]
Hi @pijyoi 

Can't tell if your issue got cut off, or you're only trying to present one option.

Those template files really drive me up the wall, wonder how feasible it would be to outright remove them, and any place that uses them in the library just loads the `.ui` file itself.

Also, given that Qt 6.1 is going to have so many more fixes/and more functionality (eventually Qt Multimedia is being rolled back in) I'm not opposed to dropping Qt 6.0 support; but to do so because of some template files seems a bit silly.

**Comment by j9ac9k** [comment]
Also thanks for keeping a lookout on this; always good to know when we might run into breakage before the breakage occurs.

**Comment by pijyoi** [comment]
>Can't tell if your issue got cut off, or you're only trying to present one option.

I could only think of one non-shim option at the time. The shim options are just extending the pain.

>Those template files really drive me up the wall, wonder how feasible it would be to outright remove them, and any place that uses them in the library just loads the .ui file itself.

On Windows, an install of Python may not put the various miscellaneous python scripts / executables within the path. Thus for PySide{2,6} which needs to launch pyside{2,6}-uic as a subprocess to generate the template file, that may not work.
Also, even when it is within the path, it takes a noticeable delay on Windows. (E.g. the examples DateAxisItem.py vs DateAxisItem_QtDesigner.py) 

Unfortunately, we can't test it out yet. Not all wheels needed to test the PyQt6 development snapshot are available.
https://www.riverbankcomputing.com/pypi/

**Comment by j9ac9k** [comment]
I thought Qt provided a mechanism to load the `.ui` files directly without having to use the UIC to make a `.py` file; need to read up more on that first.

**Comment by j9ac9k** [comment]
"Fixed" with #1707
